pandas/tests/io/formats/style/test_to_latex.py

```diff
@@ -302,6 +302,34 @@ def test_multiindex_row_and_col(df_ext):
     assert result == expected
 
 
+@pytest.mark.parametrize(
+    "multicol_align, siunitx, exp",
+    [
+        ("naive-l", False, " & A & &"),
+        ("naive-r", False, " & & & A"),
+        ("naive-l", True, "{} & {A} & {} & {}"),
+        ("naive-r", True, "{} & {} & {} & {A}"),
+    ],
+)
+def test_multicol_naive(df, multicol_align, siunitx, exp):
+    ridx = MultiIndex.from_tuples([("A", "a"), ("A", "b"), ("A", "c")])
+    df = df.astype({"A": int})
+    df.columns = ridx
+    level1 = " & a & b & c" if not siunitx else "{} & {a} & {b} & {c}"
```

Comment:
I am not a fan of having logic inside tests.
I would suggest to move ``level1`` to ``pytest.mark.parametrize`` alongside ``siunitx``.
But it may not fit nicely in one line with the other parameters, so may be leave it as it is.